aiopenapi3/extra.py

```diff
@@ -1,40 +1,61 @@
-from typing import Dict, List
+from typing import Dict, List, Union, Pattern
 import logging
+import re
 
 from .plugin import Document, Init
 
 
 class Reduced(Document, Init):
     log = logging.getLogger("aiopenapi3.extra.Reduced")
 
-    def __init__(self, operations: Dict[str, List[str]]):
+    def __init__(self, operations: Dict[Union[str, Pattern], List[str]]):
+        """Initialize the Reduced object."""
         self.operations = operations
         super().__init__()
 
     def parsed(self, ctx: "Document.Context") -> "Document.Context":
+        """Parse the given context."""
         paths = {}
-        try:
-            npi = {}
-            for path, methods in self.operations.items():
-                pi = ctx.document["paths"][path]
-                if methods is not None:
-                    npi["parameters"] = pi.get("parameters", [])
-                    for m in methods:
-                        npi[m] = pi[m]
-                else:
-                    npi = pi
-                paths[path] = npi
-            ctx.document["paths"] = paths
-        except KeyError as e:
-            self.log.exception(e)
-            raise e
+        matched_paths = self._find_matched_paths(ctx)
+
+        for path in matched_paths:
+            paths[path] = self._extract_path_info(ctx, path)
+
+        ctx.document["paths"] = paths
         return ctx
 
+    def _find_matched_paths(self, ctx: "Document.Context") -> List[str]:
+        """Find paths in the context that match the given operations."""
+        return [
+            path
+            for path_pattern in self.operations.keys()
+            for path in ctx.document["paths"].keys()
+            if (isinstance(path_pattern, str) and path_pattern == path)
+            or (isinstance(path_pattern, re.Pattern) and path_pattern.match(path))
+        ]
+
+    def _extract_path_info(self, ctx: "Document.Context", path: str) -> Dict:
+        """Extract path information from the context based on the given path."""
+        methods = self.operations[path]
```

Comment:
this tries to access a re match with the match instead of the re - does not work